crypto/merklekeystore/keystore_test.go

```diff
@@ -17,176 +17,157 @@
 package merklekeystore
 
 import (
+	"context"
 	"crypto/rand"
+	"database/sql"
+	"fmt"
+	uuid "github.com/satori/go.uuid"
 	"math"
 	"testing"
 
 	"github.com/algorand/go-algorand/crypto"
 	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/util/db"
 	"github.com/stretchr/testify/require"
 )
 
+// Is this test even needed? What is the purpose?
 func TestSignerCreation(t *testing.T) {
 	a := require.New(t)
+	var err error
 
 	h := genHashableForTest()
 	for i := uint64(1); i < 20; i++ {
-		signer, err := New(i, i+1, 1, crypto.Ed25519Type)
-		a.NoError(err)
+		signer := generateTestSigner(i, i+1, 1, a)
+		defer signer.keyStore.store.Close()
 		_, err = signer.Sign(h, i)
 		a.NoError(err)
 	}
 
-	signer, err := New(0, 0, 1, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(0, len(signer.SignatureAlgorithms))
-
-	signer, err = New(0, 1, 1, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(uint64(1), signer.FirstValid)
-	a.Equal(1, len(signer.SignatureAlgorithms))
+	testSignerNumKeysLimits := func(firstValid uint64, lastValid uint64, interval uint64, expectedLen int) {
+		signer := generateTestSigner(firstValid, lastValid, interval, a)
+		a.Equal(expectedLen, length(signer, a))
+		signer.keyStore.store.Close()
+	}
 
-	_, err = New(1, 0, 1, crypto.Ed25519Type)
-	a.Error(err)
+	testSignerNumKeysLimits(0, 0, 1, 0)
+	testSignerNumKeysLimits(0, 1, 1, 1)
+	testSignerNumKeysLimits(2, 2, 2, 1)
+	testSignerNumKeysLimits(8, 21, 10, 2)
+	testSignerNumKeysLimits(8, 20, 10, 2)
+	testSignerNumKeysLimits(10, 21, 10, 2)
+	testSignerNumKeysLimits(10, 20, 10, 2)
+	testSignerNumKeysLimits(11, 20, 10, 1)
```

Comment:
dont we want this test as well:
	_, err = New(1, 0, 1, crypto.Ed25519Type)
	a.Error(err)